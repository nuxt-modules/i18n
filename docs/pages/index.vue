<script setup lang="ts">
import { useClipboard } from '@vueuse/core'
const { data: page } = await useAsyncData('index', () => queryContent('/').findOne())

// Page Metadata (SEO & OG)
const { setPageMeta } = usePageMeta()
setPageMeta({
  title: page.value.title,
  description: page.value.description,
  headline: page.value.hero.headline?.label
})
const source = ref('npx nuxi module add i18n')
const { copy, copied } = useClipboard({ source })
</script>

<template>
  <div>
    <ULandingHero
      v-if="page.hero"
      v-bind="page.hero"
      :ui="{
        container: 'overflow-hidden py-10 flex flex-row items-center  gap-1',
        links: 'flex items-center gap-2',
        description: 'text-gray-500 dark:text-gray-400 text-xl max-w-2xl leading-normal mb-10'
      }"
    >
      <template #headline>
        <UBadge v-if="page.hero.headline" variant="subtle" size="lg" class="relative rounded-full font-semibold">
          <NuxtLink :to="page.hero.headline.to" target="_blank" class="focus:outline-none" tabindex="-1">
            <span class="absolute inset-0" aria-hidden="true" />
          </NuxtLink>

          {{ page.hero.headline?.label }}

          <UIcon
            v-if="page.hero.headline.icon"
            :name="page.hero.headline.icon"
            class="ml-1 w-4 h-4 pointer-events-none"
          />
        </UBadge>
      </template>

      <template #description>
        {{ page.hero.description }}
      </template>

      <template #title>
        <p>
          <span class="md:hidden">i18n</span>
          <span class="hidden md:block">Internationalization</span>
          for
          <span class="text-primary">Nuxt Applications</span>
        </p>
      </template>

      <UInput
        aria-label="Copy code to get started"
        :model-value="source"
        name="get-started"
        class="mx-auto"
        disabled
        autocomplete="off"
        size="lg"
        :ui="{ base: 'w-[300px] disabled:cursor-default', icon: { trailing: { pointer: '' } } }"
      >
        <template #leading>
          <UIcon name="i-ph-terminal" />
        </template>
        <template #trailing>
          <UButton
            aria-label="Copy Code"
            :color="copied ? 'green' : 'gray'"
            variant="ghost"
            :padded="false"
            :icon="copied ? 'i-ph-check' : 'i-ph-copy'"
            @click="copy(source)"
          />
        </template>
      </UInput>
    </ULandingHero>

    <ULandingSection :title="page.features.title">
      <UPageGrid>
        <ULandingCard v-for="(item, index) of page.features.items" :key="index" v-bind="item" />
      </UPageGrid>
    </ULandingSection>
  </div>
</template>
