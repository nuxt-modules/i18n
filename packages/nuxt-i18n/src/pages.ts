import createDebug from 'debug'
import { extendPages } from '@nuxt/kit'
import { localizeRoutes } from 'vue-i18n-routing'

import type { Nuxt, NuxtPage } from '@nuxt/schema'
import type { RouteOptionsResolver, ComputedRouteOptions } from 'vue-i18n-routing'
import type { NuxtI18nOptions, CustomRoutePages } from './types'

const debug = createDebug('@nuxtjs/i18n:pages')

export function setupPages(
  options: Required<NuxtI18nOptions>,
  nuxt: Nuxt,
  additionalOptions: { isBridge?: boolean; localeCodes: string[] } = {
    isBridge: false,
    localeCodes: []
  }
) {
  let includeUprefixedFallback = nuxt.options.target === 'static'
  nuxt.hook('generate:before', () => {
    debug('called generate:before hook')
    includeUprefixedFallback = true
  })

  const pagesDir = nuxt.options.dir && nuxt.options.dir.pages ? nuxt.options.dir.pages : 'pages'
  const { trailingSlash } = nuxt.options.router
  debug(`pagesDir: ${pagesDir}, tailingSlash: ${trailingSlash}`)

  extendPages(pages => {
    debug('pages making ...', pages)
    const localizedPages = localizeRoutes(pages, {
      ...options,
      includeUprefixedFallback,
      optionsResolver: getRouteOptionsResolver(options.pages, pagesDir, options.defaultLocale)
    })
    pages.splice(0, pages.length)
    pages.unshift(...(localizedPages as NuxtPage[]))
    debug('... made pages', pages)
  })
}

function getRouteOptionsResolver(
  pages: CustomRoutePages,
  pagesDir: string,
  defaultLocale: string
): RouteOptionsResolver {
  return (route, localeCodes): ComputedRouteOptions | null => {
    const options: ComputedRouteOptions = {
      locales: localeCodes,
      paths: {}
    }
    const pattern = new RegExp(`${pagesDir}/`, 'i')
    // TODO: we might be needed support for vite, this is for webpack
    const chunkName = route.chunkName ? route.chunkName.replace(pattern, '') : route.name
    const pageOptions = chunkName ? pages[chunkName] : undefined
    // Routing disabled
    if (pageOptions === false) {
      return null
    }
    // Skip if no page options defined
    if (!pageOptions) {
      return options
    }

    // Remove disabled locales from page options
    options.locales = options.locales.filter(locale => pageOptions[locale] !== false)

    // Construct paths object
    for (const locale of options.locales) {
      const customLocalePath = pageOptions[locale]
      if (typeof customLocalePath === 'string') {
        // Set custom path if any
        options.paths[locale] = customLocalePath
        continue
      }

      const customDefaultLocalePath = pageOptions[defaultLocale]
      if (typeof customDefaultLocalePath === 'string') {
        // Set default locale's custom path if any
        options.paths[locale] = customDefaultLocalePath
      }
    }

    return options
  }
}
