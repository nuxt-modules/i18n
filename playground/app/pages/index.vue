<template>
  <div>
    <h1>{{ $t('welcome') }}</h1>
  </div>
</template>
