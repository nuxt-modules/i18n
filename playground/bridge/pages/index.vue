<script setup lang="ts">
import { useI18n } from 'vue-i18n-bridge'
import { localePath, switchLocalePath, useI18nHead } from 'vue-i18n-routing'

const { t, locale, locales, getBrowserLocale } = useI18n()
const i18nHead = useI18nHead({ addSeoAttributes: true })

console.log('getBrowserLocale', getBrowserLocale())

useMeta({
  title: computed(() => t('hello', { name: 'nuxt bridge' })),
  htmlAttrs: computed(() => ({
    lang: i18nHead.value.htmlAttrs!.lang
  })),
  link: computed(() => [...(i18nHead.value.link || [])]),
  meta: computed(() => [...(i18nHead.value.meta || [])])
})

function getLocaleName(code) {
  const locale = locales.value.find(i => i.code === code)
  return locale ? locale.name : code
}

const availableLocales = computed(() => {
  return locales.value.filter(i => i.code !== locale.value)
})
</script>

<template>
  <div>
    <h1>Demo: Nuxt Bridge</h1>
    <h2>{{ $t('hello', { name: 'nuxt bridge' }) }}</h2>
    <h2>Pages</h2>
    <nav>
      <NuxtLink :to="localePath('/')">Home</NuxtLink> |
      <NuxtLink :to="localePath('/about')">About</NuxtLink>
    </nav>
    <h2>Current Language: {{ getLocaleName(locale) }}</h2>
    <h2>Select Languages</h2>
    <nav>
      <span v-for="locale in availableLocales" :key="locale.code">
        <NuxtLink :to="switchLocalePath(locale.code) || ''">{{ locale.name }}</NuxtLink> |
      </span>
    </nav>
    <h2>I18n Head</h2>
    <div>
      <p>{{ i18nHead }}</p>
    </div>
  </div>
</template>
