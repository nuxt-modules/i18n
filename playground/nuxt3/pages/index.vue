<script setup lang="ts">
import { useI18n } from 'vue-i18n'
import { localePath, switchLocalePath, useI18nHead, useBrowserLocale } from '#i18n'

const { t, locale, locales } = useI18n()

console.log('useBrowserLocale', useBrowserLocale())

const i18nHead = useI18nHead({ addSeoAttributes: true })
useMeta({
  title: computed(() => t('hello', { name: 'nuxt bridge' })),
  htmlAttrs: computed(() => ({
    lang: i18nHead.value.htmlAttrs!.lang
  })),
  link: computed(() => [...(i18nHead.value.link || [])]),
  meta: computed(() => [...(i18nHead.value.meta || [])])
})

function getLocaleName(code) {
  const locale = locales.value.find(i => i.code === code)
  return locale ? locale.name : code
}

const availableLocales = computed(() => {
  return locales.value.filter(i => i.code !== locale.value)
})
</script>

<template>
  <div>
    <h1>Demo: Nuxt 3</h1>
    <h2>{{ $t('hello', { name: 'nuxt3' }) }}</h2>
    <h2>Pages</h2>
    <nav>
      <NuxtLink :to="localePath('/')">Home</NuxtLink> |
      <NuxtLink :to="localePath('/about')">About</NuxtLink>
    </nav>
    <h2>Current Language: {{ getLocaleName(locale) }}</h2>
    <h2>Select Languages</h2>
    <nav>
      <span v-for="locale in availableLocales" :key="locale.code">
        <NuxtLink :to="switchLocalePath(locale.code) || ''">{{ locale.name }}</NuxtLink> |
      </span>
    </nav>
    <h2>I18n Head</h2>
    <div>
      <p>{{ i18nHead }}</p>
    </div>
  </div>
</template>
