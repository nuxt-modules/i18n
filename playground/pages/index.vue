<script setup lang="ts">
import { computed } from 'vue'

// import { useLocalePath, useSwitchLocalePath, useLocaleHead, useBrowserLocale } from '#i18n'
import { LocaleObject } from '#i18n'

const route = useRoute()
const { t, strategy, locale, locales, localeProperties, setLocale, finalizePendingLocaleChange } = useI18n()
const localePath = useLocalePath()
const switchLocalePath = useSwitchLocalePath()
const getRouteBaseName = useRouteBaseName()

// route.meta.pageTransition.onBeforeEnter = async () => {
//   await finalizePendingLocaleChange()
// }

console.log('route base name', getRouteBaseName())
console.log('useBrowserLocale', useBrowserLocale())
console.log('localeProperties', localeProperties)
console.log('foo', t('foo'))
console.log('message if local layer merged:', t('layerText'))
console.log('message if github layer merged:', t('layer-test-key'))

function getLocaleName(code: string) {
  const locale = (locales.value as LocaleObject[]).find(i => i.code === code)
  return locale ? locale.name : code
}

const availableLocales = computed(() => {
  return (locales.value as LocaleObject[]).filter(i => i.code !== locale.value)
})

definePageMeta({
  title: 'pages.title.top',
  middleware: () => {
    const localePath2 = useLocalePath()
    console.log('middleware', localePath2({ name: 'blog' }))
  },
  pageTransition: {
    name: 'page',
    mode: 'out-in',
    onBeforeEnter: async () => {
      const { finalizePendingLocaleChange } = useNuxtApp().$i18n
      await finalizePendingLocaleChange()
      console.log('onBeforeEnter')
    }
  }
})
</script>

<template>
  <div>
    <h1>Demo: Nuxt 3</h1>
    <h2>{{ $t('hello', { name: 'nuxt3' }) }}</h2>
    <p>{{ $t('bar.buz', { name: 'buz' }) }}</p>
    <h2>Pages</h2>
    <nav>
      <NuxtLink :to="localePath('/')">Home</NuxtLink> | <NuxtLink :to="localePath({ name: 'about' })">About</NuxtLink> |
      <NuxtLink :to="localePath({ name: 'blog' })">Blog</NuxtLink> |
      <NuxtLink :to="localePath({ name: 'category-id', params: { id: 'foo' } })">Category</NuxtLink> |
      <NuxtLink :to="localePath({ name: 'history' })">History</NuxtLink>
    </nav>
    <h2>Current Language: {{ getLocaleName(locale) }}</h2>
    <h2>Current Strategy: {{ strategy }}</h2>
    <h2>Select Languages with switchLocalePath</h2>
    <nav>
      <span v-for="locale in availableLocales" :key="locale.code">
        <NuxtLink :to="switchLocalePath(locale.code) || ''">{{ locale.name }}</NuxtLink> |
      </span>
    </nav>
    <h2>Select Languages with setLocale</h2>
    <nav>
      <span v-for="locale in availableLocales" :key="locale.code">
        <a href="javascript:void(0)" @click="setLocale(locale.code)">{{ locale.name }}</a> |
      </span>
    </nav>
    <p>{{ $t('settings.profile') }}</p>
    <p>{{ $t('tag') }}</p>
  </div>
</template>

<style scoped>
.page-enter-active,
.page-leave-active {
  transition: opacity 1s;
}
.page-enter,
.page-leave-active {
  opacity: 0;
}
</style>
