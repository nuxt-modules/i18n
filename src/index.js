import { resolve, join } from 'path'
import { readdirSync } from 'fs'
import merge from 'lodash.merge'
// @ts-ignore
import { directive as i18nExtensionsDirective } from '@intlify/vue-i18n-extensions'
import { COMPONENT_OPTIONS_KEY, DEFAULT_OPTIONS, ROOT_DIR, STRATEGIES, REDIRECT_ON_OPTIONS } from './helpers/constants'
import { buildHook, createExtendRoutesHook } from './core/hooks'
import { formatMessage } from './templates/utils-common'

/** @type {import('@nuxt/types').Module<import('../types').Options>} */
export default async function (moduleOptions) {
  /** @type {import('../types/internal').ResolvedOptions} */
  const options = merge({}, DEFAULT_OPTIONS, moduleOptions, this.options.i18n)

  if (!Object.values(STRATEGIES).includes(options.strategy)) {
    // eslint-disable-next-line no-console
    console.error(formatMessage(`Invalid "strategy" option "${options.strategy}" (must be one of: ${Object.values(STRATEGIES).join(', ')}).`))
    return
  }

  if (options.lazy && !options.langDir) {
    throw new Error(formatMessage('When using the "lazy" option you must also set the "langDir" option.'))
  }

  if (options.langDir) {
    if (!options.locales.length || typeof options.locales[0] === 'string') {
      throw new Error(formatMessage('When using the "langDir" option the "locales" must be a list of objects.'))
    }
    for (const locale of options.locales) {
      if (typeof (locale) === 'string' || !locale.file) {
        throw new Error(formatMessage(`All locales must be objects and have the "file" property set when using "langDir".\nFound none in:\n${JSON.stringify(locale, null, 2)}.`))
      }
    }
    options.langDir = this.nuxt.resolver.resolveAlias(options.langDir)
  }

  // Templates (including plugins).
  // This is done here rather than in the build hook to ensure the order the plugins are added
  // is predictable between different modules.
  const nuxtOptions = this.options
  const normalizedLocales = []
  for (const locale of options.locales) {
    if (typeof (locale) === 'string') {
      normalizedLocales.push({ code: locale })
    } else {
      normalizedLocales.push(locale)
    }
  }
  options.normalizedLocales = normalizedLocales
  // Get an array of locale codes from the list of locales.
  options.localeCodes = options.normalizedLocales.map(locale => locale.code)

  const templatesOptions = {
    Constants: {
      COMPONENT_OPTIONS_KEY,
      STRATEGIES,
      REDIRECT_ON_OPTIONS
    },
    nuxtOptions: {
      isUniversalMode: nuxtOptions.mode === 'universal',
      trailingSlash: nuxtOptions.router.trailingSlash
    },
    options
  }

  const templatesPath = join(__dirname, '/templates')
  for (const file of readdirSync(templatesPath)) {
    if (!file.endsWith('.js')) {
      continue
    }
    if (file.startsWith('plugin.')) {
      this.addPlugin({
        src: resolve(templatesPath, file),
        fileName: join(ROOT_DIR, file),
        options: templatesOptions
      })
    } else {
      this.addTemplate({
        src: resolve(templatesPath, file),
        fileName: join(ROOT_DIR, file),
        options: templatesOptions
      })
    }
  }

  if (options.strategy !== STRATEGIES.NO_PREFIX && options.localeCodes.length) {
    this.extendRoutes(createExtendRoutesHook.call(this, options))
  }

  await this.nuxt.hook('build:before', () => buildHook.call(this, options))

  this.options.alias['~i18n-klona'] = require.resolve('klona/full').replace(/\.js$/, '.mjs')
  this.options.alias['~i18n-ufo'] = require.resolve('ufo').replace(/\.js$/, '.mjs')

  if (!Array.isArray(this.options.router.middleware)) {
    throw new TypeError(formatMessage('options.router.middleware is not an array.'))
  }
  this.options.router.middleware.push('nuxti18n')

  if (!this.options.render.bundleRenderer || typeof (this.options.render.bundleRenderer) !== 'object') {
    throw new TypeError(formatMessage('options.render.bundleRenderer is not an object.'))
  }
  this.options.render.bundleRenderer.directives = this.options.render.bundleRenderer.directives || {}
  this.options.render.bundleRenderer.directives.t = i18nExtensionsDirective

  // Transpile is-https (IE11)
  this.options.build.transpile = this.options.build.transpile || []
  this.options.build.transpile.push('is-https')
}
