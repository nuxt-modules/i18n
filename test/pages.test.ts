import { vi, describe, test, expect } from 'vitest'
import { localizeRoutes } from 'vue-i18n-routing'
import { getRouteOptionsResolver } from '../src/pages'
import fs from 'node:fs'

import type { NuxtI18nOptions } from '../src/types'
import type { NuxtHooks } from '@nuxt/schema'

type ExtractArrayType<T> = T extends (infer U)[] ? U : never
type NuxtPage = ExtractArrayType<Parameters<NuxtHooks['pages:extend']>[0]>

function getNuxtOptions(pages: Required<NuxtI18nOptions>['pages'], parsePages = false): NuxtI18nOptions {
  return {
    parsePages,
    pages,
    defaultLocale: 'en',
    strategy: 'prefix_except_default',
    defaultLocaleRouteNameSuffix: 'default',
    trailingSlash: false,
    routesNameSeparator: '___',
    locales: [
      { code: 'en', iso: 'en-US', file: 'en.json', name: 'English' },
      { code: 'ja', iso: 'ja-JP', file: 'ja.json', name: 'Japanses' },
      { code: 'fr', iso: 'fr-FR', file: 'fr.json', name: 'FranÃ§ais' }
    ]
  }
}

test('basic', async () => {
  vi.spyOn(fs, 'readFileSync').mockReturnValue('')

  const options = getNuxtOptions({})
  const pages: NuxtPage[] = [
    {
      path: '/about',
      file: '/path/to/nuxt-app/pages/about.vue',
      children: [
        {
          name: 'about',
          path: '',
          file: '/path/to/nuxt-app/pages/about/index.vue',
          children: []
        }
      ]
    },
    {
      name: 'index',
      path: '/',
      file: '/path/to/nuxt-app/pages/index.vue',
      children: []
    }
  ]
  const localizedPages = localizeRoutes(pages, {
    ...options,
    includeUprefixedFallback: false,
    optionsResolver: getRouteOptionsResolver('pages', options as Required<NuxtI18nOptions>)
  })
  expect(localizedPages).toMatchSnapshot()
})

describe('custom route', () => {
  describe('configuration', () => {
    test('simple', async () => {
      vi.spyOn(fs, 'readFileSync').mockReturnValue('')

      const options = getNuxtOptions({ about: { ja: '/about-ja' } })
      const pages: NuxtPage[] = [
        {
          path: '/about',
          file: '/path/to/nuxt-app/pages/about.vue',
          children: [
            {
              name: 'about',
              path: '',
              file: '/path/to/nuxt-app/pages/about/index.vue',
              children: []
            }
          ]
        },
        {
          name: 'index',
          path: '/',
          file: '/path/to/nuxt-app/pages/index.vue',
          children: []
        }
      ]
      const localizedPages = localizeRoutes(pages, {
        ...options,
        includeUprefixedFallback: false,
        optionsResolver: getRouteOptionsResolver('pages', options as Required<NuxtI18nOptions>)
      })
      expect(localizedPages).toMatchSnapshot()
    })

    test('nested static route', async () => {
      vi.spyOn(fs, 'readFileSync').mockReturnValue('')

      const options = getNuxtOptions({
        'nested/route': {
          en: '/mycustompath/nested/route'
        }
      })
      const pages: NuxtPage[] = [
        {
          name: 'nested-route',
          path: '/nested/route',
          file: '/path/to/nuxt-app/pages/nested/route/index.vue',
          children: []
        }
      ]
      const localizedPages = localizeRoutes(pages, {
        ...options,
        includeUprefixedFallback: false,
        optionsResolver: getRouteOptionsResolver('pages', options as Required<NuxtI18nOptions>)
      })
      expect(localizedPages).toMatchSnapshot()
    })

    test('nested dynamic route', async () => {
      vi.spyOn(fs, 'readFileSync').mockReturnValue('')

      const options = getNuxtOptions({
        ':nested/:route': {
          en: '/mycustompath/:nested/:route'
        },
        ':nested/:route/:slug(.*)*': {
          en: '/mycustompath/:nested/:slug(.*)*'
        }
      })
      const pages: NuxtPage[] = [
        {
          name: 'nested-route-slug',
          path: '/:nested/:route/:slug(.*)*',
          file: '/path/to/nuxt-app/pages/[nested]/[route]/[...slug].vue',
          children: []
        },
        {
          name: 'nested-route',
          path: '/:nested/:route',
          file: '/path/to/nuxt-app/pages/[nested]/[route]/index.vue',
          children: []
        }
      ]
      const localizedPages = localizeRoutes(pages, {
        ...options,
        includeUprefixedFallback: false,
        optionsResolver: getRouteOptionsResolver('pages', options as Required<NuxtI18nOptions>)
      })
      expect(localizedPages).toMatchSnapshot()
    })

    test('nested complex route', async () => {
      const options = getNuxtOptions({
        about: {
          fr: '/a-propos'
        },
        services: {
          fr: '/offres'
        },
        'services/development': {
          fr: '/offres/developement'
        },
        'services/development/app': {
          fr: '/offres/developement/app'
        },
        'services/development/website': {
          fr: '/offres/developement/site-web'
        },
        'services/coaching': {
          fr: '/offres/formation'
        }
      })
      const pages: NuxtPage[] = [
        {
          name: 'about',
          path: '/about',
          file: '/path/to/nuxt-app/pages/about/index.vue',
          children: []
        },
        {
          name: 'services-development-app',
          path: '/services/development/app',
          file: '/path/to/nuxt-app/pages/services/development/app/index.vue',
          children: []
        },
        {
          name: 'services-development-coaching',
          path: '/services/development/coaching',
          file: '/path/to/nuxt-app/pages/services/development/coaching/index.vue',
          children: []
        },
        {
          name: 'services-development',
          path: '/services/development',
          file: '/path/to/nuxt-app/pages/services/development/index.vue',
          children: []
        },
        {
          name: 'services-development-website',
          path: '/services/development/website',
          file: '/path/to/nuxt-app/pages/services/development/website/index.vue',
          children: []
        },
        {
          name: 'services',
          path: '/services',
          file: '/path/to/nuxt-app/pages/services/index.vue',
          children: []
        }
      ]
      const localizedPages = localizeRoutes(pages, {
        ...options,
        includeUprefixedFallback: false,
        optionsResolver: getRouteOptionsResolver('pages', options as Required<NuxtI18nOptions>)
      })
      expect(localizedPages).toMatchSnapshot()
    })
  })

  describe.todo('component', () => {
    // TODO:
  })
})
