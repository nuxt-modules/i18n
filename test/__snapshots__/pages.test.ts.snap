// Vitest Snapshot v1

exports[`basic 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "file": "/path/to/nuxt-app/pages/about/index.vue",
        "name": "about___en",
        "path": "",
      },
    ],
    "file": "/path/to/nuxt-app/pages/about.vue",
    "path": "/about",
  },
  {
    "children": [
      {
        "children": [],
        "file": "/path/to/nuxt-app/pages/about/index.vue",
        "name": "about___ja",
        "path": "",
      },
    ],
    "file": "/path/to/nuxt-app/pages/about.vue",
    "path": "/ja/about",
  },
  {
    "children": [
      {
        "children": [],
        "file": "/path/to/nuxt-app/pages/about/index.vue",
        "name": "about___fr",
        "path": "",
      },
    ],
    "file": "/path/to/nuxt-app/pages/about.vue",
    "path": "/fr/about",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/index.vue",
    "name": "index___en",
    "path": "/",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/index.vue",
    "name": "index___ja",
    "path": "/ja",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/index.vue",
    "name": "index___fr",
    "path": "/fr",
  },
]
`;

exports[`custom route > component > dynami route 1`] = `
[
  {
    "children": [],
    "name": "articles-name___en",
    "path": "/articles/:name",
  },
  {
    "children": [],
    "name": "articles-name___ja",
    "path": "/ja/記事/:name",
  },
  {
    "children": [],
    "name": "articles-name___fr",
    "path": "/fr/articles/:name",
  },
]
`;

exports[`custom route > component > script-setup 1`] = `
[
  {
    "children": [],
    "path": "/about-us",
  },
  {
    "children": [],
    "path": "/ja/about-ja",
  },
  {
    "children": [],
    "path": "/fr/a-propos",
  },
]
`;

exports[`custom route > component > simple 1`] = `
[
  {
    "children": [],
    "path": "/about-us",
  },
  {
    "children": [],
    "path": "/ja/about-ja",
  },
  {
    "children": [],
    "path": "/fr/a-propos",
  },
]
`;

exports[`custom route > configuration > nested complex route 1`] = `
[
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/about/index.vue",
    "name": "about___en",
    "path": "/about",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/about/index.vue",
    "name": "about___ja",
    "path": "/ja/about",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/about/index.vue",
    "name": "about___fr",
    "path": "/fr/a-propos",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/app/index.vue",
    "name": "services-development-app___en",
    "path": "/services/development/app",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/app/index.vue",
    "name": "services-development-app___ja",
    "path": "/ja/services/development/app",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/app/index.vue",
    "name": "services-development-app___fr",
    "path": "/fr/offres/developement/app",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/coaching/index.vue",
    "name": "services-development-coaching___en",
    "path": "/services/development/coaching",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/coaching/index.vue",
    "name": "services-development-coaching___ja",
    "path": "/ja/services/development/coaching",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/coaching/index.vue",
    "name": "services-development-coaching___fr",
    "path": "/fr/services/development/coaching",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/index.vue",
    "name": "services-development___en",
    "path": "/services/development",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/index.vue",
    "name": "services-development___ja",
    "path": "/ja/services/development",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/index.vue",
    "name": "services-development___fr",
    "path": "/fr/offres/developement",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/website/index.vue",
    "name": "services-development-website___en",
    "path": "/services/development/website",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/website/index.vue",
    "name": "services-development-website___ja",
    "path": "/ja/services/development/website",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/website/index.vue",
    "name": "services-development-website___fr",
    "path": "/fr/offres/developement/site-web",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/index.vue",
    "name": "services___en",
    "path": "/services",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/index.vue",
    "name": "services___ja",
    "path": "/ja/services",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/index.vue",
    "name": "services___fr",
    "path": "/fr/offres",
  },
]
`;

exports[`custom route > configuration > nested dynamic route 1`] = `
[
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/[nested]/[route]/[...slug].vue",
    "name": "nested-route-slug___en",
    "path": "/mycustompath/:nested/:slug(.*)*",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/[nested]/[route]/[...slug].vue",
    "name": "nested-route-slug___ja",
    "path": "/ja/mycustompath/:nested/:slug(.*)*",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/[nested]/[route]/[...slug].vue",
    "name": "nested-route-slug___fr",
    "path": "/fr/mycustompath/:nested/:slug(.*)*",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/[nested]/[route]/index.vue",
    "name": "nested-route___en",
    "path": "/mycustompath/:nested/:route",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/[nested]/[route]/index.vue",
    "name": "nested-route___ja",
    "path": "/ja/mycustompath/:nested/:route",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/[nested]/[route]/index.vue",
    "name": "nested-route___fr",
    "path": "/fr/mycustompath/:nested/:route",
  },
]
`;

exports[`custom route > configuration > nested static route 1`] = `
[
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/nested/route/index.vue",
    "name": "nested-route___en",
    "path": "/mycustompath/nested/route",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/nested/route/index.vue",
    "name": "nested-route___ja",
    "path": "/ja/mycustompath/nested/route",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/nested/route/index.vue",
    "name": "nested-route___fr",
    "path": "/fr/mycustompath/nested/route",
  },
]
`;

exports[`custom route > configuration > simple 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "file": "/path/to/nuxt-app/pages/about/index.vue",
        "name": "about___en",
        "path": "",
      },
    ],
    "file": "/path/to/nuxt-app/pages/about.vue",
    "path": "/about",
  },
  {
    "children": [
      {
        "children": [],
        "file": "/path/to/nuxt-app/pages/about/index.vue",
        "name": "about___ja",
        "path": "/ja/about-ja",
      },
    ],
    "file": "/path/to/nuxt-app/pages/about.vue",
    "path": "/ja/about-ja",
  },
  {
    "children": [
      {
        "children": [],
        "file": "/path/to/nuxt-app/pages/about/index.vue",
        "name": "about___fr",
        "path": "",
      },
    ],
    "file": "/path/to/nuxt-app/pages/about.vue",
    "path": "/fr/about",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/index.vue",
    "name": "index___en",
    "path": "/",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/index.vue",
    "name": "index___ja",
    "path": "/ja",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/index.vue",
    "name": "index___fr",
    "path": "/fr",
  },
]
`;
